mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -942,21 +999,75 @@ func expandAutoHealingPolicies(configured []interface{}) []*compute.InstanceGrou
 	return autoHealingPolicies
 }
 
-func expandStatefulPolicy(configured []interface{}) *compute.StatefulPolicy {
+func expandStatefulPolicy(d *schema.ResourceData) *compute.StatefulPolicy {
+	preservedState := &compute.StatefulPolicyPreservedState{
+	}
+	stateful_disks := d.Get("stateful_disk").(*schema.Set).List()
 	disks := make(map[string]compute.StatefulPolicyPreservedStateDiskDevice)
-	for _, raw := range configured {
+	for _, raw := range stateful_disks {
 		data := raw.(map[string]interface{})
 		disk := compute.StatefulPolicyPreservedStateDiskDevice{
 			AutoDelete: data["delete_rule"].(string),
 		}
 		disks[data["device_name"].(string)] = disk
 	}
-	if len(disks) > 0 {
-		return &compute.StatefulPolicy{PreservedState: &compute.StatefulPolicyPreservedState{Disks: disks}}
+	preservedState.Disks = disks
+	var change bool
+	change = false
+	<% unless version == "ga" -%>
+	if d.HasChange("stateful_internal_ip") {
+		oldInternalIps, newInternalIps := d.GetChange("stateful_internal_ip")
+		if len(newInternalIps.(*schema.Set).List()) == 0 {
+			preservedState.NullFields = append(preservedState.NullFields, "internalIPs")
+		} else {
+			preservedState.InternalIPs = expandStatefulIps(oldInternalIps.(*schema.Set).List(), newInternalIps.(*schema.Set).List(), preservedState, "InternalIPs.%s")
+		}
+		change = true
+	}
+	if d.HasChange("stateful_external_ip") {
+		oldExternalIps, newExternalIps := d.GetChange("stateful_external_ip")
+		if len(newExternalIps.(*schema.Set).List()) == 0 {
+			preservedState.NullFields = append(preservedState.NullFields, "externalIPs")
+		} else {
+			preservedState.ExternalIPs = expandStatefulIps(oldExternalIps.(*schema.Set).List(), newExternalIps.(*schema.Set).List(), preservedState, "ExternalIPs.%s")
+		}
+		change = true
+	}
+	<% end -%>
+	if len(disks) > 0 || change {
```

Comment:
I see this maintains the old behavior (always send disks if len > 0) and I'm a little hesitant about suggesting changing that - but also, if the preservedState.NullFields approach is preferred for stateful ips, maybe we should actually be doing that with stateful disks as well? What do you think?

Reply:
Or alternately we could make the new fields match this behavior? Basically it seems like they should probably be consistent?

Reply:
I'm a little hesitant in making changes to existing logic. I'll take a deeper look and see if I can come up with something.

Reply:
I tried with NullFields approach and didn't notice a difference. I prefer leaving it as it is for now.